pkg/xds/envoy/types.go

```diff
@@ -82,7 +84,39 @@ func DistinctTags(tags []Tags) []Tags {
 	return result
 }
 
-type Clusters map[string][]ClusterSubset
+type Cluster struct {
+	subsets            []ClusterSubset
+	hasExternalService bool
+	requireTls         bool
+}
+
+func (c *Cluster) Add(subset ClusterSubset) {
+	c.subsets = append(c.subsets, subset)
+	if subset.IsExternalService {
+		c.hasExternalService = true
+	}
+	if subset.RequiresTls {
+		c.requireTls = true
+	}
+}
+
+func (c *Cluster) Tags() []Tags {
+	var result []Tags
+	for _, info := range c.subsets {
+		result = append(result, info.Tags)
+	}
+	return result
+}
+
+func (c *Cluster) HasExternalService() bool {
```

Comment:
I don't think you can mix and match external service subsets with regular subsets, therefore this should be `IsExternalService()`